mlir/lib/ExecutionEngine/CMakeLists.txt

```diff
@@ -181,6 +181,7 @@ if(LLVM_ENABLE_PIC)
   add_mlir_library(mlir_arm_sme_abi_stubs
     SHARED
     ArmSMEStubs.cpp)
+  target_compile_definitions(mlir_arm_sme_abi_stubs PRIVATE mlir_arm_sme_abi_stubs_EXPORTS)
```

Comment:
The case is surprising to me, is this something we do in the other libs?

Reply:
To work on Windows, yes. There are some ways to tell it to indiscriminately export everything but I don't think we use that.

Reply:
`LLVM_ATTRIBUTE_WEAK` is expanding to empty macro on Windows, which resulted dll without any exported symbols, and export lib is not being generated for such dll (I don't know if it's general cmake behavior, llvm cmake or microsoft linker behavior). Adding `__declspec(dllexport)` fixed it. We don't see this for other libs because all of them export some symbols.